frontend/lib/Views/AdminPage/members/members_view.dart

```diff
@@ -0,0 +1,252 @@
+            floatingActionButton: _buildAddButton(),
+          ),
+        ),
+      );
+
+  FutureBuilder _buildBody() => FutureBuilder(
+        future: widget.gateway.getGroupData(widget.groupId).onError(onError),
+        builder: (context, snapshot) {
+          if (snapshot.connectionState == ConnectionState.none ||
+              snapshot.data == null) {
+            return loadingInCenter();
+          } else {
+            membersListProvider.makeMemberList(snapshot.data!);
+            return SingleChildScrollView(
+              controller: ScrollController(),
+              child: Consumer<MembersListProvider>(
+                builder: (context, __, _) => Column(
+                  children: [
+                    _buildSortBar(membersListProvider),
+                    _buildMembersList(membersListProvider.membersList.length)
+                  ],
+                ),
+              ),
+            );
+          }
+        },
+      );
+
+  Container _buildSortBar(MembersListProvider membersListProvider) => Container(
+        margin: const EdgeInsets.only(top: 20),
+        alignment: Alignment.centerLeft,
+        padding: const EdgeInsets.symmetric(horizontal: 30),
+        child: Row(
+          children: [
+            ToggleButtons(
+              selectedColor: Colors.red,
+              isSelected: widget._selections,
+              fillColor: Colors.transparent,
+              renderBorder: false,
+              children: [
+                Container(
+                  margin: const EdgeInsets.only(left: 20, right: 20),
+                  child: Text(
+                    "ID",
+                    style: TextStyle(
+                      fontFamily: 'SofiaSans',
+                      fontSize: 25,
+                      fontWeight: FontWeight.normal,
+                      color: membersListProvider.idColor,
+                    ),
+                  ),
+                ),
+                Container(
+                  margin: const EdgeInsets.only(left: 20, right: 20),
+                  child: Text(
+                    "EMAIL",
+                    style: TextStyle(
+                      fontFamily: 'SofiaSans',
+                      fontSize: 25,
+                      fontWeight: FontWeight.normal,
+                      color: membersListProvider.emailColor,
+                    ),
+                  ),
+                )
+              ],
+              onPressed: (int index) {
+                membersListProvider.changeSorting(index);
+              },
+            )
+          ],
+        ),
+      );
+
+  ListView _buildMembersList(int itemCount) => ListView.builder(
+        controller: ScrollController(),
+        itemCount: itemCount,
+        shrinkWrap: true,
+        padding: const EdgeInsets.symmetric(horizontal: 20),
+        itemBuilder: (context, i) =>
+            _buildGroupCard(membersListProvider.membersList[i]),
+      );
+
+  Card _buildGroupCard(MemberInfo member) => Card(
```

Comment:
Zastanowiłbym się, czy nie lepiej było wrzucić budowanie karty do osobnego `stateless` widgetu, tym samym przerzucić go do osobnego pliku, bo ten już trochę wydaje się za duży

Reply:
czy ja wiem, wszystkie nasze widoki mają po 250 linii

Reply:
Okej, to w takim wypadku nie musisz tego ruszać, taka była moja tylko sugestia